auth/auth0_manager.go

```diff
@@ -1,23 +1,95 @@
 package auth
 
 import (
-	"log"
+	"fmt"
+	"time"
 
+	"github.com/zpatrick/go-cache"
 	"github.com/zpatrick/rclient"
 )
 
 type Auth0Manager struct {
-	token  string
-	client *rclient.RestClient
+	clientID   string
+	connection string
+	client     *rclient.RestClient
+	cache      *cache.Cache
 }
 
-func NewAuth0Manager(endpoint, token string) *Auth0Manager {
+type Auth0Config struct {
+	Domain     string
+	ClientID   string
+	Connection string
+}
+
+type oauthReq struct {
+	ClientID   string `json:"client_id"`
+	Connection string `json:"connection"`
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	GrantType  string `json:"grant_type"`
+	Scope      string `json:"scope"`
+}
+
+type authStatus struct {
```

Comment:
Given it's an internal implementation detail, you wouldn't test this object itself. 
We should test what the `Auth0Manager` does (validates creds, doesn't DDOS auth0), not how it does it (as much as possible).